src/coreclr/vm/i386/jithelp.asm

```diff
@@ -1214,39 +1214,6 @@ JIT_TailCallVSDLeave:
 
 JIT_TailCall ENDP
 
-
-;------------------------------------------------------------------------------
-
-; HCIMPL2_VV(float, JIT_FltRem, float dividend, float divisor)
-@JIT_FltRem@8 proc public
-        fld  dword ptr [esp+4]          ; divisor
-        fld  dword ptr [esp+8]          ; dividend
-fremloop:
-        fprem
-        fstsw   ax
-        fwait
-        sahf
-        jp      fremloop        ; Continue while the FPU status bit C2 is set
-        fxch    ; swap, so divisor is on top and result is in st(1)
-        fstp    ST(0)           ; Pop the divisor from the FP stack
-        retn    8               ; Return value is in st(0)
-@JIT_FltRem@8 endp
-
-; HCIMPL2_VV(float, JIT_DblRem, float dividend, float divisor)
-@JIT_DblRem@16 proc public
-        fld  qword ptr [esp+4]          ; divisor
-        fld  qword ptr [esp+12]         ; dividend
-fremloopd:
-        fprem
-        fstsw   ax
-        fwait
-        sahf
-        jp      fremloopd       ; Continue while the FPU status bit C2 is set
-        fxch    ; swap, so divisor is on top and result is in st(1)
-        fstp    ST(0)           ; Pop the divisor from the FP stack
-        retn    16              ; Return value is in st(0)
-@JIT_DblRem@16 endp
```

Comment:
Hmmm, I see 6x perf regression on my ad-hoc microbenchmark from removing this asm implementation. 6x perf regression sounds like a lot. Do you see the same?

```
using System;
using System.Diagnostics;
using System.Runtime.CompilerServices;

for (;;)
{
    var sw = new Stopwatch();
    sw.Start();
    Work(12345678);
    Console.WriteLine(sw.ElapsedMilliseconds);
}

[MethodImpl(MethodImplOptions.NoInlining)]
double Work(double x)
{
   double sum = 0;
   for (double i = 100; i < 100000000.0; i++)
       sum += x % i;
   return sum;
}
```

Reply:
I am keeping the x86 asm helpers for now.

Reply:
I assume it's either the FPU to SSE transitions that @tannergooding mentioned before or you're hitting the slow path in the native implementation that calls into the FPU anyway. Do you see the same for floats too? Those don't have the slow path.

Reply:
> Do you see the same for floats too? 

Yes, I see the same regression with floats too.

> I assume it's either the FPU to SSE transitions that @tannergooding mentioned before

Where was this mentioned?

I know that the matrix of the Intel floating-point transition costs is super complicated and there are probably situations that have not regressed.

I have tried to approximate the simplest real-world use in ad-hoc micro-benchmarks: call the method with some varying input and interleave it with some other simple floating-point computation. If this simplest case shows very large regression, I do not see the point in digging further.

Reply:
> Where was this mentioned?

It was mentioned on the C# Discord when we discussed the 25x regression I've observed with the AVX512 path for double casts. Tanner mentioned there that since we pass FP in the FPU on x86, it might trigger costly transitions in the native helpers that do operations with SSE, especially for variants that are VEX encoded. When I asked why does the path exist then if its so costly, Tanners answer was that we have an edge case where every helper call uses the FPU and normal C++ code won't see it as much (the MSVC helper uses a custom calling convention that passes it in vector registers IIRC).